monkeytype/cli.py

```diff
@@ -127,7 +127,10 @@ def run_handler(args: argparse.Namespace, stdout: IO, stderr: IO) -> None:
     sys.argv = sys.argv[2:]
     try:
         with trace(args.config):
-            runpy.run_path(args.script_path, run_name='__main__')
+            if args.m:
+                runpy.run_module(args.script_path, run_name='__main__', alter_sys=True)
```

Comment:
do we want `run_name='__main__'`? Won't that make traces from the named module unusable?

OTOH, without it we won't trigger `if __name__ == '__main__'` blocks.

Reply:
Yes. And then there's actually another problem too. Should there be no import guard inside `__main__.py`, then `monkeytype stub foobar.__main__` launches the application 😦. Maybe this isn't going to be easily workable.

Opinions?

Reply:
On the other hand, `python -m` runs it as `__main__` as well anyways. Maybe it makes sense to rewrite away `__main__` when saving the trace?

Reply:
I don't have great ideas here. This is why I didn't implement support for this originally; it seemed clearer to maintain a clean separation between the script that exercises the code (which can't be monkeytyped) and the code itself (which can).

I guess the best option, if we're going to do it, is to merge it like this, with the caveat that whatever you run as the script won't be typed. In that case, needs some docs here.

Reply:
Hmm, that's a good point. I don't have a problem with rewriting `__main__` to the module name when saving a trace. With #19 I wasn't sure how we'd figure out the right module name, but here the user already gave it to us, so that's not an issue. If you get that working, with docs and tests, I'd happily merge.

Reply:
So I thought about this a bit more and there are still some problems but also some potential solutions. We don't really know the module name to rewrite it to, since it could be one of: `foo.py`, `foo.__init__.py`, or `foo.__main__.py`, depending on the module.

We could figure it out with some probing of the filesystem, and possibly some other tricks I haven't thought of, but might not be easily workable.

However another option might be to do it entry points style, e.g. something like the explicit `foo.__main__:main`, and instead of using `run_module` we import `main` from `foo.__main__` and run it ourselves.

Reply:
I'm inclined towards what you have now, which is the simplest and least magical implementation, exactly parallel to `python -m`, and just documenting its limitations.

The entry-point style approach would actually be more useful for a package which uses entry points. So you'd give something like `-e foo:main` and we'd internally run the equivalent of:

```
from foo import main
with monkeytype.trace():
    main()
```

This seems like a separate feature from `-m` support. I'm open to having both.